CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationRewards.uc

```diff
@@ -231,3 +257,215 @@ static protected function bool RewardNotAvaliable(optional XComGameState NewGame
 	// Since these rewards are only used for display purposes, we flag them as unavaliable to prevent P1/P2 CAs from randomly spawning
 	return false;
 }
+
+///////////////////
+/// POI rewards ///
+///////////////////
+
+static function X2DataTemplate CreateLootTableRewardTemplate (name RewardName, name LootTableName)
+{
+	local X2RewardTemplate Template;
+
+	`CREATE_X2Reward_TEMPLATE(Template, RewardName);
+	Template.rewardObjectTemplateName = LootTableName;
+	
+	Template.SetRewardByTemplateFn = SetLootTableReward;
+	Template.GiveRewardFn = GiveLootTableReward;
+	Template.GetRewardStringFn = GetLootTableRewardString;
+
+	return Template;
+}
+
+static function SetLootTableReward(XComGameState_Reward RewardState, name TemplateName)
+{
+	RewardState.RewardObjectTemplateName = TemplateName;
+}
```

Comment:
How/when does this work?

Reply:
I dunno, it's taken from the basegame LootTableReward. It works, that's all I know

Reply:
`RewardState.RewardObjectTemplateName` is not referenced anywhere else, hence I wondered how it would work (if it was called anywhere)

Reply:
I remember now that the FindGlobalLootCarrier function in X2LootManager checks for the RewardState's attached object and uses that as the carrier to search for.

Reply:
Never mind, the object is just passed as a regular name to that function in GiveLootTableReward